packages/x-data-grid/src/hooks/features/overlays/useGridOverlays.tsx

```diff
@@ -8,7 +8,7 @@ import { gridPinnedRowsCountSelector } from '../rows/gridRowsSelector';
 import { GridLoadingOverlayVariant } from '../../../components/GridLoadingOverlay';
 import { GridOverlayWrapper } from '../../../components/base/GridOverlays';
 import type { GridOverlayType } from '../../../components/base/GridOverlays';
-import { gridVisibleColumnDefinitionsSelector } from '..';
```

Comment:
It would be ideal to prevent introducing such imports in the first place. 🙈 
Could we look into adding an ESLint rule for it? 🤔